src/stepfunctions/steps/constants.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from enum import Enum
+from stepfunctions.steps.fields import Field
+
+# Path to SageMaker placeholder parameters
+placeholder_paths = {
+    # Paths taken from https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateProcessingJob.html
+    'ProcessingStep': {
+        Field.Role.value: ['RoleArn'],
+        Field.ImageUri.value: ['AppSpecification', 'ImageUri'],
+        Field.InstanceCount.value: ['ProcessingResources', 'ClusterConfig', 'InstanceCount'],
+        Field.InstanceType.value: ['ProcessingResources', 'ClusterConfig', 'InstanceType'],
+        Field.Entrypoint.value: ['AppSpecification', 'ContainerEntrypoint'],
+        Field.VolumeSizeInGB.value: ['ProcessingResources', 'ClusterConfig', 'VolumeSizeInGB'],
+        Field.VolumeKMSKey.value: ['ProcessingResources', 'ClusterConfig', 'VolumeKmsKeyId'],
+        Field.Env.value: ['Environment'],
+        Field.Tags.value: ['Tags'],
```

Comment:
thought: is there a way to read these from the SDK or automate it? having this hand-rolled can be problematic for a few reasons:

1. drift if the API signatures expand
2. prone to error, since it's reliant on everything being hand rolled

Reply:
I had the same thought - doing this by hand can introduce errors and is not easily maintainable.

Another option would be, for each SageMaker property, to call the function that adds it to the Parameters in the Sagemaker code instead of manually  getting the path from the [API docs](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateProcessingJob.html)
That would mean:
- Each property will call a different function (if existing) in sagemaker in order to add it to Parameters
- Some properties will not have an existing function in sagemaker that adds them to Parameters and we will have to do it by hand using the `placeholder_paths`

Reply:
Read @wong-a 's proposed solution after posting the previous comment - will go with that since it removes the need to map each args with `placeholder_paths`